### PR TITLE
feat: plot decision tree

### LINKED ISSUE #856: Plot decision tree
### Is your feature request related to a problem?

Decision trees are one of the few models that are easy to visualize. However, we don't offer any visualization at the moment.

### Desired solution

* Add a method `plot() -> Image` to `DecisionTreeRegressor` and `DecisionTreeClassifier`.
* Raise a `ModelNotFittedError` if the model is not fitted yet.

Implementation:
* Use [`plot_tree`](https://scikit-learn.org/stable/modules/generated/sklearn.tree.plot_tree.html).
* Avoid code duplication, e.g. by adding the method to the common superclass `_DecisionTreeBase`.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by timwirt**
Closes #856 

### Summary of Changes

- Added plot method for DecisionTreeClassifier and DecisionTreeRegressor.
- Added tests for both

### CODE REVIEW

File: tests/safeds/ml/classical/classification/__snapshots__/test_decision_tree/TestPlot.test_should_check_that_plot_image_is_same_as_plt_figure.png
Comment (lars-reimann) [inline tests/safeds/ml/classical/classification/__snapshots__/test_decision_tree/TestPlot.test_should_check_that_plot_image_is_same_as_plt_figure.png:1]: Did you encounter cases where a box hid text of another box due to overlapping?

Comment (timwirt) [reply tests/safeds/ml/classical/classification/__snapshots__/test_decision_tree/TestPlot.test_should_check_that_plot_image_is_same_as_plt_figure.png:1]: Not until now

File: src/safeds/ml/classical/regression/_decision_tree_regressor.py [hunk 1]
```diff
@@ -71,3 +73,36 @@ def _get_sklearn_model(self) -> RegressorMixin:
...
+
+    def plot(self) -> Image:
+        """
+        Get the image of the decision tree.
+
+        Returns
+        -------
+        Image:
+            The decision tree figure as an image.
+        """
```
Comment (lars-reimann) [inline src/safeds/ml/classical/regression/_decision_tree_regressor.py:89]: Add `Raises` section to highlight that the method raises if the model is not fitted.

File: src/safeds/ml/classical/classification/_decision_tree_classifier.py [hunk 2]
```diff
@@ -71,3 +73,36 @@ def _get_sklearn_model(self) -> ClassifierMixin:
...
+
+    def plot(self) -> Image:
+        """
+        Get the image of the decision tree.
+
+        Returns
+        -------
+        Image:
+            The decision tree figure as an image.
+        """
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_decision_tree_classifier.py:89]: Add `Raises` section to highlight that the method raises if the model is not fitted.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds/ml/classical/regression/_decision_tree_regressor.py (hunk 1 above)
Comment (timwirt) [reply src/safeds/ml/classical/regression/_decision_tree_regressor.py:89]: Fixed

File: src/safeds/ml/classical/classification/_decision_tree_classifier.py (hunk 2 above)
Comment (timwirt) [reply src/safeds/ml/classical/classification/_decision_tree_classifier.py:89]: Fixed

**Review by lars-reimann — APPROVED** [review]
👍

**Pull request merged**